lib/dotcom_web/components/system_status/subway_status.ex

```diff
@@ -141,18 +141,45 @@ defmodule DotcomWeb.Components.SystemStatus.SubwayStatus do
     |> Enum.flat_map(&rows_for_route(&1, include_alert: true))
   end
 
-  defp maybe_collapse_rows(rows) when length(rows) > @max_rows, do: collapse_rows(rows)
-  defp maybe_collapse_rows(rows), do: rows
+  defp collapse_rows_if_needed(rows) do
+    rows
+    |> if_too_large(&collapse_identical_route_rows/1)
+    |> if_too_large(&collapse_disrupted_green_line_rows/1)
+    |> if_too_large(&collapse_all_green_line_rows/1)
```

Comment:
Note: This implicitly preserves Mattapan rows because none of ☝️ functions will ever collapse a Mattapan row into a Red row.